### ISSUE TITLE
validate-table problems

### ISSUE THREAD

**Issue by wwood** [biocore/biom-format#696]
Hi,

I'm trying to test my biom installation by generating and validating biom tables, but I'm having some issues. 

```
$ python
Python 2.7.4 (default, Apr 12 2013, 14:17:44) 
[GCC 4.6.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from biom.util import biom_open  
>>> from biom.table import Table
>>> from numpy import array
>>> t = Table(array([[1, 2], [3, 4]]), ['a', 'b'], ['x', 'y'])
>>> 
>>> with biom_open('foo.biom', 'w') as f: t.to_hdf5(f, "example")
... 
>>> with open('foo.json.biom', 'w') as f: t.to_json('me', f)
... 
>>> 
```

then validating

```
$ biom validate-table -i foo.biom
Unknown BIOM type: 
The input file is not a valid BIOM-formatted file.
```

```
$ biom validate-table -i foo.json.biom 
Traceback (most recent call last):
  File "/srv/sw/py274_modules/biom-format/2.1.5/bin/biom", line 9, in <module>
    load_entry_point('biom-format==2.1.5', 'console_scripts', 'biom')()
  File "/srv/sw/py274_modules/click/6.6/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/srv/sw/py274_modules/click/6.6/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/srv/sw/py274_modules/click/6.6/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/srv/sw/py274_modules/click/6.6/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/srv/sw/py274_modules/click/6.6/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/srv/sw/py274_modules/biom-format/2.1.5/lib/python2.7/site-packages/biom/cli/table_validator.py", line 49, in validate_table
    valid, report = _validate_table(input_fp, format_version, detailed_report)
  File "/srv/sw/py274_modules/biom-format/2.1.5/lib/python2.7/site-packages/biom/cli/table_validator.py", line 62, in _validate_table
    detailed_report=detailed_report)
  File "/srv/sw/py274_modules/biom-format/2.1.5/lib/python2.7/site-packages/biom/cli/table_validator.py", line 113, in __call__
    detailed_report=detailed_report)
  File "/srv/sw/py274_modules/biom-format/2.1.5/lib/python2.7/site-packages/biom/cli/table_validator.py", line 97, in run
    return self._validate_json(**kwargs)
  File "/srv/sw/py274_modules/biom-format/2.1.5/lib/python2.7/site-packages/biom/cli/table_validator.py", line 305, in _validate_json
    status_msg = method(table_json)
  File "/srv/sw/py274_modules/biom-format/2.1.5/lib/python2.7/site-packages/biom/cli/table_validator.py", line 496, in _valid_type
    if value.lower() not in self.TableTypes:
AttributeError: 'NoneType' object has no attribute 'lower'
```

json seems at least well formed:

```
$ cat foo.json.biom 
{"id": "None","format": "Biological Observation Matrix 1.0.0","format_url": "http://biom-format.org","generated_by": "me","date": "2016-05-12T20:24:05.433281","matrix_element_type": "float","shape": [2, 2],"type": null,"matrix_type": "sparse","data": [[0,0,1.0],[0,1,2.0],[1,0,3.0],[1,1,4.0]],"rows": [{"id": "a", "metadata": null},{"id": "b", "metadata": null}],"columns": [{"id": "x", "metadata": null},{"id": "y", "metadata": null}]}
```

I'm going to take a stab and say that `validate-table` isn't quite working, not the table generation part, and that maybe it isn't a problem with my install? Can someone confirm whether this works on their system please?

Thanks.

**Comment by wasade** [comment]
It has to deal with one of the subtle differences between BIOM 1.0.0 and 2.x. In 1.0.0, the "table_type" key is always written, whereas it is not always written by the 2.x formatter, leading to `"None"` returned in the former and `None` returned in the latter. Thanks for reporting this issue.

**Comment by wasade** [comment]
fixed in #733
